src/features/vote/ActiveRequestsForm.tsx

```diff
@@ -245,15 +279,20 @@ const ActiveRequestsForm: FC<Props> = ({
                 <td>
                   <div>
                     {/* <UnlockedIcon /> */}
-                    {votePhase === "Commit" && el.vote
+                    {votePhase === "Commit" && el.vote !== UNDEFINED_VOTE
                       ? "Committed"
-                      : votePhase === "Commit" && !el.vote
+                      : votePhase === "Commit"
                       ? "Uncommitted"
                       : null}
-                    {votePhase === "Reveal" && el.vote
+                    {votePhase === "Reveal" && el.revealed
+                      ? "Revealed"
+                      : votePhase === "Reveal" &&
+                        el.vote !== UNDEFINED_VOTE &&
+                        !el.revealed
                       ? "Reveal"
-                      : /* this is incorrect -- just wanna hide this string for now. */
-                        null}
+                      : votePhase === "Reveal" && el.vote === UNDEFINED_VOTE
+                      ? "Uncommitted"
+                      : null}
```

Comment:
Thoughts on an alternative approach where you do the following above the JSX:
```js
tableValues.map((el, index) => {
  let stateEntry;
  let voteEntry;
  switch(votePhase) {
    case "Commit":
      // Logic
      break;
    case "Reaveal":
      // Logic
      break;
  }
}
// .. return JSX down here
```

There's certainly a tradeoff here -- the logic is placed away from where it's displayed. I only raise the point because nested ternaries are pretty hard to follow, but there doesn't seem to be a good alternative when the logic is inside the JSX.

Reply:
Yeah it's more a stylistic choice.

To be honest, because of how complex this ActiveRequest form view *is* in general, it is tempting to split off the commit and reveal states into their own views in general. There are potentially enough differences between the views on my chatting with Tom that this is probably okay. To be honest, the "reveal" stage, this is actually no longer a form, not in the sense we'd think about in an HTML sense.

Reply:
Agreed on all of the above. I'm not super strongly opinionated. My gut response to seeing nested ternaries is that it's hard to read. But I totally get the constraints of JSX. Just wanted to mention it, but seems like you've already given it a decent amount of thought.